apps/login/src/lib/zitadel.ts

```diff
@@ -300,6 +313,41 @@ export async function getUserByID(userId: string) {
   return userService.getUserByID({ userId }, {});
 }
 
+export async function verifyInviteCode(
+  userId: string,
+  verificationCode: string,
+) {
+  return userService.verifyInviteCode({ userId, verificationCode }, {});
+}
+
+export async function resendInviteCode(userId: string) {
+  return userService.resendInviteCode({ userId }, {});
+}
+
+export async function createInviteCode(userId: string, host: string | null) {
+  let medium = create(SendInviteCodeSchema, {
+    applicationName: "Typescript Login",
+  });
+
+  if (host) {
+    medium = {
+      ...medium,
+      urlTemplate: `${host.includes("localhost") ? "http://" : "https://"}${host}/verify?code={{.Code}}&userId={{.UserID}}&organization={{.OrgID}}&invite=true`,
```

Comment:
can't we read the whole origin from the context instead of assuming https if we are not on localhost?

Reply:
The protocol can be read from other request headers, I think Proto, X-Forwarded-Proto, X-Forwarded or so. Let's leave it like this for now.